silo/gviews_v4.py

```diff
@@ -214,64 +232,52 @@ def import_from_gsheet_helper(user, silo_id, silo_name, spreadsheet_id,
     headers = []
     lvss = []
 
-    for r, row in enumerate(data):
-        if r == 0:
-            headers = []
-            for header in row:
-                header = cleanKey(header)
-                headers.append(header)
+    # get the column names
+    header = values.pop(0)
+    for h in header:
+        h = cleanKey(h)
+        headers.append(h)
 
-            addColsToSilo(silo, headers)
-            continue
-        filter_criteria = {}
+    data = _convert_gsheet_data(headers, values)
 
+    col_type, data = get_column_types(data)
+    for r, row in enumerate(data):
+        filter_criteria = {}
         # build filter_criteria if unique field(s) have been setup for this silo
-        for unique_field in unique_fields:
+        for uf in unique_fields:
             try:
-                filter_criteria.update(
-                    {unique_field.name: row[headers.index(
-                        unique_field.name)].strip()})
+                filter_criteria.update({str(uf.name): str(row[uf.name])})
             except KeyError:
                 pass
```

Comment:
This is a dangerous practice, can you find another way to validate the fields instead?

Reply:
I didn't get your comment. Which validation? We're only building the filter.

Reply:
- I realized that this is wrong. It should raise `AttributeError` instead.
- Why is it a dangerous practice? The Python cultural norm says that in many cases, you must use exceptions for control-flow.
- I don't know another way to validate the fields. There is no function like `has_attribute` and the method `__get_attribute__` raises the `AttributeError` exception as well.

Reply:
Ok, dangerous might have been a too strong word. Of course you can use exceptions, but if we just pass them i think it might cause problems if there is an unexpected case of this exception and we pass it without further notifications.
I took another look at it and think it's ok in this case. But at least it pointed you to the wrong exception type.